app/jobs/invoices/abacus/create_yearly_abo_alpen_invoices_job.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+class Invoices::Abacus::CreateYearlyAboAlpenInvoicesJob < RecurringJob
+  SLICE_SIZE = 25  # number of people/invoices transmitted per abacus batch request
+
+  def perform_internal
+    process_invoices
+  end
+
+  def next_run
+    # Sets next run to 02:18 of next day
+    Time.zone.tomorrow.at_beginning_of_day.change(hour: 2, minute: 18).in_time_zone
+  end
+
+  def error(job, exception)
+    create_error_log_entry("stapelverarbeitung", "Jahresrechnungen Abo Magazin Die Alpen konnten nicht an Abacus übermittelt werden. " \
+              "Es erfolgt ein weiterer Versuch.", exception.message)
+  end
+
+  def failure(job)
+    create_error_log_entry("stapelverarbeitung", "Rollierendes Inkassolauf Abo Magazin Die Alpen abgebrochen.", nil)
+  end
+
+  private
+
+  def active_abonnenten
+    Role.joins(:person)
+      .left_joins(person: :external_invoices)
+      .where.not(people: {abacus_subject_key: nil})
+      .where(type: Group::AboMagazin::Abonnent.sti_name, terminated: false)
+      .where(end_on: Time.zone.today..62.days.from_now)
+      .where("external_invoices.id IS NULL OR external_invoices.year != EXTRACT(YEAR FROM roles.end_on + INTERVAL '1 day')")
```

Comment:
Hier muss noch die Prüfung auf den external_invoices.type `ExternalInvoice::AboMagazin` mit rein. Wahrscheinlich allerdings via NOT EXISTS Subquery, da im Falle von mehreren Rechnungen sonst ein Abonnent dennoch mit einbezogen wird.

Testcase: Person hat folgende Rechnungen, Jahr 2025
- ExternalInvoice::AboMagazin von 2024
- ExternalInvoice::CourseParticipation von 2025

=> Person soll im Result enthalten sein

- Zusätzlich zu oben ExternalInvoice::AboMagazin von 2025

=> Person soll nicht im Resultat enthalten sein.